apollo-router/src/plugins/connectors/plugin.rs

```diff
@@ -45,11 +47,14 @@ impl Plugin for Connectors {
         let debug_extensions = init.config.debug_extensions
             || std::env::var(CONNECTORS_DEBUG_ENV).as_deref() == Ok("true");
 
-        if debug_extensions && LOG_ONE_TIME.get().is_none() {
-            tracing::warn!(
-                "Connector debugging is enabled, this may expose sensitive information."
-            );
-            LOG_ONE_TIME.get_or_init(|| ());
+        {
+            if debug_extensions && *LAST_DEBUG_ENABLED_VALUE.lock() != debug_extensions {
+                tracing::warn!(
+                    "Connector debugging is enabled, this may expose sensitive information."
+                );
+            }
+
+            *LAST_DEBUG_ENABLED_VALUE.lock() = debug_extensions;
         }
```

Comment:
```suggestion
        let last_value = LAST_DEBUG_ENABLED_VALUE.load(Ordering::Relaxed);
        if LAST_DEBUG_ENABLED_VALUE
            .compare_exchange(
                last_value,
                debug_extensions,
                Ordering::Relaxed,
                Ordering::Relaxed,
            )
            .is_ok()
            && debug_extensions
        {
            tracing::warn!(
                "Connector debugging is enabled, this may expose sensitive information."
            );
        }
```